### ISSUE TITLE
ament_lint WMD track

### ISSUE THREAD

**Issue by tylerjw** [ros-planning/moveit2#375]
This is for tracking the work of fixing up linting issues so the moveit2 codebase passes linters from [ament_lint](https://github.com/ament/ament_lint).

## What do I need to know to help?

`ament_lint` is a collection of python scripts that can be added as tests to ros2 packages that run various linters over the package.  Each one has two separate packages, `ament_<linter>` and `ament_cmake_<linter>`.  The first one contains the script that runs the test and the second contains the cmake macro for enabling it.  There is also another one called [`ament_lint_auto`](https://github.com/ament/ament_lint/blob/master/ament_lint_auto/doc/index.rst) that has some cmake magic to make running the default version of various ament lint tests easier.  It works by setting up each ament_lint test that it finds in your `package.xml`.  Lastly, there is `ament_lint_common` that includes a set of linters decided on by the core ros2 team.  When you depend on this package in your package.xml you get these linters:

```
ament_cmake_copyright
ament_cmake_cppcheck
ament_cmake_cpplint
ament_cmake_flake8
ament_cmake_lint_cmake
ament_cmake_pep257
ament_cmake_uncrustify
ament_cmake_xmllint
```

[This PR I created](https://github.com/ros-planning/moveit2/pull/340) enables these tests through our packages and uses a bit of cmake magic to disable tests that don't pass yet.

## What should I do?

Building off that PR (hopefully this is merged into main before we start this) remove one of magic lines in the cmake that disables a test, run the test, and fixup lint issues it finds in our code, then submit a PR with your changes to make that test pass.

## What about `clang_format`?

We will not be making any changes that we cannot get clang_format to pass with (this means that some of the linters we may not be able to enable).  To run clang_format locally to see if your changes will pass do this:

1. Create a new workspace on your machine to build `ament_lint` from source.  (The most recent version released to foxy does not contain some features needed to use `ament_clang_format` for moveit2).

```bash
mkdir -p ~/ws_ros2_tools/src
cd ~/ws_ros2_tools/src
git clone git@github.com:ament/ament_lint.git
```

2. Build and source that workspace

```bash
cd ~/ws_ros2_tools
colcon build
source install/local_setup.bash
```

3. Return to your moveit2 workspace and use ament_clang_tidy to test if your local changes will pass clang_format.

```bash
cd ~/ws_moveit2/src/moveit2
ament_clang_format --clang-format-version 10 --config .clang-format
```

## How do I get help?

I will be on discord here: https://discord.gg/psGvKguh and here: https://discord.gg/vMd9AzC4 during wmd to help people do this.

**Comment by tylerjw** [comment]
One thing to note is we are *NOT* enabling `ament_uncrustiy` for any package because it conflicts with `clang_format`

**Comment by tylerjw** [comment]
We did a bunch of this on WMD.  Closing this issue for now.
